pkg/controller.v1/tensorflow/pod.go

```diff
@@ -70,7 +70,23 @@ func (tc *TFController) reconcilePods(
 
 	initializeTFReplicaStatuses(tfjob, rtype)
 
-	podSlices := tc.GetPodSlices(pods, replicas, logger)
+	podSlices, podsToBeRemoved := tc.GetPodSlices(pods, replicas, logger)
+
+	// Scale down
+	if tfjob.Spec.EnableDynamicWorker && len(podsToBeRemoved) > 0 {
+		// Currently only allow to scale down workers
+		if rtype == tfv1.TFReplicaTypeWorker {
+			logger.Infof("Removing %d workers", len(podsToBeRemoved))
+			for _, pods := range podsToBeRemoved {
+				if err := tc.PodControl.DeletePod(tfjob.Namespace, pods.Name, tfjob); err!=nil {
```

Comment:
Prefer to have a event `tc.Recorder.Eventf`

Reply:
You mean to have it for error, or for pod deletion event?

Reply:
The latter.